packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -264,18 +270,25 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
     isCodeEditorExpandedFocused,
     Boolean(documentationSections),
     Boolean(editorIsInline),
-    isHistoryOpen
+    isHistoryOpen,
+    !!hideHeaderWhenExpanded
   );
   const isDark = isDarkMode;
   const editorModel = useRef<monaco.editor.ITextModel>();
   const editor1 = useRef<monaco.editor.IStandaloneCodeEditor>();
   const containerRef = useRef<HTMLElement>(null);
 
-  const editorClassName = classNames('TextBasedLangEditor', {
-    'TextBasedLangEditor--expanded': isCodeEditorExpanded,
-    'TextBasedLangEditor--compact': isCompactFocused,
-    'TextBasedLangEditor--initial': !isCompactFocused,
-  });
+  const editorClassName = classNames(
+    'TextBasedLangEditor',
+    css`
+      max-width: 100%;
+    `,
```

Comment:
I'm using `@emotion/css` here, but that is a first for this file, so I'm also happy to change to something else.

Reply:
Do you want to apply it instead on the https://github.com/elastic/kibana/blob/main/packages/kbn-text-based-editor/src/text_based_languages_editor.tsx#L809

```
css={css`
    max-width: 100%;
`}
```

also import from emotion react as we do in other places too

`import { css } from '@emotion/react';`

Just for consistency reasons